src/pages/StakingOptionBalancePage.tsx

```diff
@@ -92,8 +98,63 @@ function BalanceDetails() {
       navigate("/balance");
     } catch (err) {}
   }, [soBalanceDetails, stakeValue, connection, wallet, name]);
+
+  const handleLiquidateClick = useCallback(async () => {
+    if (!connection || !soBalanceDetails || !wallet) {
+      return;
+    }
+    const { base, soName, lotSize, baseAtoms } = soBalanceDetails;
+    const so = new StakingOptions(connection.rpcEndpoint);
+    const soStateObj = await so.getState(soName, base);
+    const liquidateParams: any = {
+      // @ts-ignore
+      strike: soStateObj.strikes[0].toNumber(),
+      soName,
+      // @ts-ignore
+      soBaseMint: soStateObj.baseMint.toBase58(),
+      userWallet: wallet.publicKey.toBase58(),
+    };
+    if (stakeValue) {
+      const amount = Math.round(
+        (Number(stakeValue) / lotSize) * 10 ** baseAtoms
+      );
+      liquidateParams.amount = amount;
+    }
+    console.log("PARAMS", liquidateParams);
+    let signature = "";
+    fetch(`${DUAL_API_MAINNET}/orders/liquidateso`, {
+      method: "post",
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(liquidateParams),
+    })
+      .then(async (data) => {
+        const buffer = await data.json();
+        const recoveredTransaction = Transaction.from(
+          Buffer.from(buffer, "base64")
+        );
+        // @ts-ignore
+        const signedTx = await wallet.signTransaction(recoveredTransaction);
+        signature = await connection.sendRawTransaction(signedTx.serialize(), {
+          skipPreflight: true,
+        });
+        console.log("signature:", signature);
+        await queryClient.invalidateQueries(["balance/so", wallet.publicKey]);
```

Comment:
should we be waiting for confirmation before invalidating and bouncing?

Reply:
Same logic as handleExerciseClick() above. Defer to @diegofigs